### PR TITLE
Adding cypress + example test

### PR DESCRIPTION
**Pull request by Tamir198**
## Check List

- [x] The title of my pull request is a short description of the changes
- [ ] This PR relates to some issue: 
- [x] I have documented the changes made (if applicable)
- [x] I have covered the changes with unit tests

Added cypress, and some first component test

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -75,7 +76,8 @@
     "lint": "npx eslint src/",
     "lint:fix": "npm run lint -- --fix",
     "prettier": "npx prettier src/ --check",
-    "prettier:fix": "npm run prettier -- --write"
+    "prettier:fix": "npm run prettier -- --write",
+    "cypress": "cypress open"
```
Comment (nir2002) [inline frontend/package.json:80]:
```suggestion
    "cypress:open": "cypress open",
    "cypress:run": "cypress run"
```

File: frontend/src/components/Button.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import Button from "./Button";
```
Comment (nir2002) [inline frontend/src/components/Button.cy.tsx:1]: it seems like example components, lets our own real components!

File: frontend/src/components/Button.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+describe("Button component tests", () => {
+  it("renders", () => {
+    cy.mount(<Button onClick={() => {}}>Click me</Button>);
+    cy.get("button").should("exist");
```
Comment (nir2002) [inline frontend/src/components/Button.cy.tsx:6]: when you test our button, better to check text is rendering as well

File: frontend/src/components/TextInput.cy.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import TextInput from "./TextInput";
```
Comment (nir2002) [inline frontend/src/components/TextInput.cy.tsx:1]: Also, we need to test our components

**Review by nir2002 — CHANGES_REQUESTED** [review]

File: frontend/src/components/Button.cy.tsx (hunk 2 above)
Comment (Tamir198) [reply frontend/src/components/Button.cy.tsx]: Changed - now the tet is using the project button

File: frontend/src/components/Button.cy.tsx (hunk 3 above)
Comment (Tamir198) [reply frontend/src/components/Button.cy.tsx:9]: Done

File: frontend/src/components/TextInput.cy.tsx (hunk 4 above)
Comment (Tamir198) [reply frontend/src/components/TextInput.cy.tsx]: Changed to use our TextInput

File: frontend/package.json (hunk 1 above)
Comment (Tamir198) [reply frontend/package.json]: Changed this to what you suggestes

**Review by nir2002 — APPROVED** [review]

**Pull request merged**